src/shared/typescriptLambdaHandlerSearch.ts

```diff
@@ -35,59 +37,200 @@ export class TypescriptLambdaHandlerSearch implements LambdaHandlerSearch {
+                        handlerName: `${baseFilename}.${exportNode.name.getText()}`,
                     })
                 }
             }
-        }
+        })
 
         return handlers
     }
 
+    /**
+     * @description Whether or not the given expression is attempting to assign to '[module.]exports.foo'
+     * @param expressionStatement Expression node to evaluate
+     */
+    private static isModuleExportsAssignment(expressionStatement: ts.ExpressionStatement): boolean {
+        if (ts.isBinaryExpression(expressionStatement.expression)) {
+            const lhsComponents: string[] = expressionStatement.expression.left.getText().split('.').map(x => x.trim())
+
+            return (lhsComponents.length === 3 && lhsComponents[0] === 'module' && lhsComponents[1] === 'exports')
+                || (lhsComponents.length === 2 && lhsComponents[0] === 'exports')
+        }
+
+        return false
+    }
+
+    /**
+     * @description Whether or not the given expression appears to be assigning a candidate Lambda Handler
+     * Expression could be one of:
+     *      [module.]exports.foo = alreadyDeclaredFunction
+     *      [module.]exports.foo = (event, context) => { ... }
+     * @param expressionStatement Expression node to evaluate
+     * @param functionHandlerNames Names of declared functions considered to be Handler Candidates
+     */
+    private static isEligibleLambdaHandlerAssignment(
+        expressionStatement: ts.ExpressionStatement,
+        functionHandlerNames: string[]
+    ): boolean {
+        if (ts.isBinaryExpression(expressionStatement.expression)) {
+            return this.isTargetFunctionReference(expressionStatement.expression.right, functionHandlerNames)
+                || this.isValidFunctionAssignment(expressionStatement.expression.right)
+        }
+
+        return false
+    }
+
+    /**
+     * @description Whether or not the given expression appears to contain a function of interest on the right hand side
+     *
+     * Example expression:
+     *      something = alreadyDeclaredFunction
+     * @param expression Expression node to evaluate
+     * @param targetFunctionNames Names of functions of interest
+     */
+    private static isTargetFunctionReference(expression: ts.Expression, targetFunctionNames: string[]): boolean {
+        if (ts.isIdentifier(expression)) {
+            return (targetFunctionNames.indexOf(expression.text) !== -1)
+        }
+
+        return false
+    }
+
+    /**
+     * @description Whether or not the given expression appears to have a function that could be a valid Lambda handler
+     * on the right hand side.
+     *
+     * Example expression:
+     *      something = (event, context) => { }
+     * @param expression Expression node to evaluate
+     * @param targetFunctionNames Names of functions of interest
+     */
+    private static isValidFunctionAssignment(expression: ts.Expression): boolean {
+        if (ts.isFunctionLike(expression)) {
```

Comment:
Just to clarify, does `isFunctionLike` support cases like `= function() { /* ... */ }`? [The whole API is undocumented](https://github.com/Microsoft/TypeScript/blob/eb2297df022f6b1f284aff96e93c06097bb01ea5/lib/typescript.d.ts), which makes it hard to review 😢